File: 2024-2/0682/unidad04.e - Statement 03/SoftPubPersonaDA/src/pe/edu/pucp/softpub/persona/dao/EspecialidadDAO.java
package pe.edu.pucp.softpub.persona.dao;

import java.util.ArrayList;
import pe.edu.pucp.softpub.persona.model.Especialidad;

public interface EspecialidadDAO {
    
    public Integer insertar(Especialidad especialidad);
    
    public Integer modificar(Especialidad especialidad);
    
    public Integer eliminar(Especialidad especialidad);
    
    public ArrayList<Especialidad> listarTodos();
    
    public Especialidad obtenerPorId(String idEspecialidad);
}

File: 2024-2/0684/unidad06.d_Web_04/SoftPub/SoftPubPersonaControler/DAO/EspecialidadDAO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubPersonaModel;

namespace SoftPubPersonaControler.DAO
{
    public interface EspecialidadDAO
    {        
        int insertar(Especialidad especialidad);

        int modificar(Especialidad especialidad);

        int eliminar(Especialidad especialidad);

        BindingList<object> listarTodos();

        Especialidad obtenerPorId(int idEspecialidad);        
    }
}
